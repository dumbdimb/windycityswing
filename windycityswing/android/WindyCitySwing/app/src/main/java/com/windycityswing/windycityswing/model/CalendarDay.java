package com.windycityswing.windycityswing.model;

import java.util.ArrayList;
import java.util.List;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class CalendarDay {
    private int day;
    private List<Dance> dances;

    public void addDance(Dance dance) {
        if (dances == null) {
            dances = new ArrayList<>();
        }
        dances.add(dance);
    }
}