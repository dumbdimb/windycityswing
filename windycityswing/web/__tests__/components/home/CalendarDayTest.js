import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';

import { Router } from 'react-router';
import createHistory from 'history/createBrowserHistory';

import CalendarDay from '..\\..\\..\\src\\scripts\\components\\home\\CalendarDay';

describe('CalendarDay', () => {
    const testEvent = {
        "title": "Test Event",
        "className": "test-event",
        "date": {
            "year": 2017,
            "month": 5,
            "day": 1,
            "weekday": 3
        },
        "timezone": {
            "startTimestamp": "20170501T210000",
            "endTimestamp": "20170501T220000"
        },
        "colors": {
            "bg": "#AAAAAA",
            "text": "black"
        }
    };
    let tree;

    beforeEach(() => {
        tree = shallow(<CalendarDay day={1} events={[testEvent]} month={5}/>);
    });

    it('renders correctly', () => {
        expect(renderer.create(<Router history={createHistory()}><CalendarDay day={1} events={[testEvent]} month={5}/></Router>).toJSON()).toMatchSnapshot();
    });

    describe('rendering logic', () => {
        describe('when day is not undefined', () => {
            const tree = shallow(<CalendarDay day={1}/>);

            it('has className calendar-day', () => {
                expect(tree.props().className).toEqual('calendar-day');
            });

            it('shows what day it is', () => {
                expect(tree.text()).toEqual('1');
            });
        });

        describe('when day is invalid', () => {
            it('does not render when prop is not specified', () => {
                const tree = shallow(<CalendarDay/>);
                expect(tree.text()).toEqual('');
            });

            it('does not render when day is undefined', () => {
                const tree = shallow(<CalendarDay day={undefined}/>);
                expect(tree.text()).toEqual('');
            });
        });

        describe('when event is undefined', () => {
            it('does not render anything event related', () => {
                tree = shallow(<CalendarDay day={1}/>);
                expect(tree.find('.calendar-day__event').length).toEqual(0);
            });
        });

        describe('when event is specified', () => {
            it('renders the event', () => {
                expect(tree.find('.calendar-day__event').length).not.toEqual(0);
            });
        })
    });
});