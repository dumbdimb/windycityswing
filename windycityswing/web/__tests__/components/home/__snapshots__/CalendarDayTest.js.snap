// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarDay renders correctly 1`] = `
<td
  className="calendar-day"
  onClick={[Function]}
>
  <span>
    1
  </span>
  <div
    className="calendar-day__events"
  >
    <div
      className="mdHidden calendar-day__event"
      style={
        Object {
          "backgroundColor": "#AAAAAA",
        }
      }
    />
    <span
      className="xsHidden calendar-day__event"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#AAAAAA",
          "color": "black",
        }
      }
    >
      Test Event
      <div
        className="calendar-day__event-hover-layer"
      >
        Test Event
      </div>
    </span>
  </div>
</td>
`;
